lib/ansiblelint/utils.py

```diff
@@ -732,3 +739,89 @@ def get_rule_skips_from_line(line):
         noqa_text = line.split('# noqa')[1]
         rule_id_list = noqa_text.split()
     return rule_id_list
+
+
+def is_playbook(filename):
+    try:
+        f = parse_yaml_from_file(filename)
+        if isinstance(f, AnsibleSequence) and hasattr(f[0], 'hosts'):
+            return True
+    except Exception as e:
+        print(
+            "Warning: Failed to load %s with %s, assuming is not a playbook."
+            % (filename, e))
+    return False
+
+
+def get_playbooks_and_roles(verbose=False):
+
+    # git is preferred as it also considers .gitignore
+    files = OrderedDict.fromkeys(sorted(subprocess.check_output(
+        ["git", "ls-files", "*.yaml", "*.yml"],
+        universal_newlines=True).split()))
+
+    playbooks = []
+    role_dirs = []
+    role_internals = set(['defaults', 'vars', 'tasks', 'handlers', 'meta', 'files', 'templates'])
+    exclude_dirs = set([
```

Comment:
Same here + I don't think these should be hardcoded right here. This needs a config with a default value.

Reply:
In fact we already have `exclude_paths` which are applied after this detection but I will reuse them. Since I introduced the is_playbook there is no danger in parsing all, so I could remove these from here. We share the sentiment about that hardcoded list.